src/main/java/org/broadinstitute/hellbender/tools/copynumber/formats/collections/SimpleCountCollection.java

```diff
@@ -86,6 +124,51 @@ private static SimpleCountCollection readHDF5(final HDF5File file) {
         return new SimpleCountCollection(metadata, simpleCounts);
     }
 
+    /**
+     * Read all counts from a Google Cloud Storage URL.
+     * A corresponding index for the counts file must also be present.
+     */
+    public static SimpleCountCollection readFromGCS(final String path) {
+        IOUtils.assertFileIsReadable(IOUtils.getPath(path));
+        Utils.validate(BucketUtils.isCloudStorageUrl(path), "Read-count path must be a Google Cloud Storage URL.");
+        Utils.validate(new SimpleCountCodec().canDecode(path), String.format(
+                "Read-count file extension must be one of the following: [%s]",
+                String.join(",", SimpleCountCodec.SIMPLE_COUNT_CODEC_EXTENSIONS)));
+        return readOverlappingSubsetFromGCS(path, null);  //specifying a null intervalSubset returns all counts
+    }
+
+    /**
+     * From a Google Cloud Storage URL, subset only the counts with intervals overlapping with intervals from a given list.
+     * A corresponding index for the counts file must also be present.
+     * The given list may contain intervals that do not overlap with any count intervals.
+     * Unlike {@link #readAndSubset(File, Set)}, this method checks for overlaps, rather than requiring the intervals
+     * to be strictly coincident.  Calling code can use {@link IntervalUtils#getIntervalsWithFlanks} to create a merged
+     * list of the original intervals desired to be strictly coincident; this merged list can then be used with this method.
+     * @param overlapIntervals    if {@code null} or empty, all counts will be returned; must be sorted and non-overlapping otherwise
+     */
+    public static SimpleCountCollection readOverlappingSubsetFromGCS(final String path,
```

Comment:
I would LOVE for all code paths for reading counts use the same code path with `FeatureDataSource`, as you have done here. I guess this isn't possible currently with HDF5... but we should be able to use `FeatureDataSource` for local TSVs as well. Don't need to address for this PR but this should be a goal.

Reply:
Yes, let's consider phasing out HDF5 count files.  They really only give a significant speedup for WGS somatic PoN building; it might be possible to achieve this by just optimizing TableReader instead.  See https://github.com/broadinstitute/gatk/issues/2858#issuecomment-317614503 for some old numbers.